File: uke2/opg205-oversettelse-av-klasse-fra-python-til-java/Brev.java
import java.util.ArrayList;

class Brev {
    private String avsender;
    private String mottaker;
    private ArrayList<String> linjer;
    
    public Brev(String avsenderen, String mottakeren) {
	avsender = avsenderen;
	mottaker = mottakeren;
	linjer = new ArrayList<String>();
    }

    public void skrivLinje(String linje) {
	linjer.add(linje);
    }

    public void lesBrev() {
	System.out.println("Hei, " + mottaker);
	System.out.println();
	for (String linje : linjer) {
	    System.out.println(linje);
	}
	System.out.println();
	System.out.println("Hilsen fra, ");
	System.out.println(avsender);
    }
}

File: uke2/opg205-oversettelse-av-klasse-fra-python-til-java/c++/Brev.cpp
#include "Brev.h"
#include <iostream>

Brev::Brev(String avsender, String mottaker) :
  m_avsender(std::move(avsender)), m_mottaker(std::move(mottaker)) { }

void Brev::skrivLinje(String linje) {
  m_linjer.push_back(linje);
}

void Brev::lesBrev() {
  std::cout << "Hei, " << m_mottaker << "." << std::endl;
  std::cout << std::endl;
  for (String linje : m_linjer) {
    std::cout << linje << std::endl;
  }
  std::cout << std::endl;
  std::cout << "Hilsen fra, " << std::endl;
  std::cout << m_avsender << std::endl;
}
